system/lib/wasmfs/wasmfs.cpp

```diff
@@ -27,8 +27,11 @@ __attribute__((init_priority(100))) WasmFS wasmFS;
 # 28 "wasmfs.cpp"
 
 std::shared_ptr<Directory> WasmFS::initRootDirectory() {
-  auto rootDirectory = std::make_shared<Directory>(S_IRUGO | S_IXUGO | S_IWUGO);
-  auto devDirectory = std::make_shared<Directory>(S_IRUGO | S_IXUGO);
+  addBackend(createMemoryFileBackend());
```

Comment:
How about we have `addBackend` return a `backend_t` and we call it from _inside_ `createMemoryFileBackend()`, which can also return a `backend_t`. That way `createMemoryFileBackend()` will have a C-compatible interface and exposing it to user code will be simpler. It would also let us get rid of `getBackend`, which doesn't make much sense in its current form because backends should always be referenced by `backend_t`, which is not a table index (for now).